### DISCUSSION TITLE
有无比较方便的更新方案

### DISCUSSION THREAD

**Discussion by jiang-yuan** [hooke007/mpv_PlayKit#117] [Q&A 问答] [answered]
我用的去年9月的版本,有些设置自己改了
直接覆盖好像会覆盖我的设置
一个个筛选对比复制到新的包下面也比较麻烦,毕竟是懒人包,有没有懒人的更新方案?

另外求一个真人电影好用的补帧方案

**Comment by hooke007** [comment]
没有，而且覆盖更新是错误的，这是我一直在release里强调的。
正确的做法是，移除旧的，解压新的。

懒人包的目的是方便二次自定义和简化学习成本。mpv的文件结构总体还不算复杂，个人要mod的几个设置基本都是conf文件（深度修改那就另当别论了）

这是个一直以来很尴尬的问题，我鼓励客制化自己的mpv，就像mpv本身的特性那样——懒人包的某个版本就是你的最终版本，它最终会完成他的价值使命，你培养定型了属于自己的mpv使用习惯。
如果紧跟着我版本更新，那大概率没法做大幅改动，你需要自己记录修改的内容然后同步我的更新，也需要对mpv了解程度足够高（随着了解深入最终还是走上了上面那条路），要么彻底不管。

**Reply by hooke007** [reply]
至于真人电影补帧那个，可选的是 mvtools-Pot，或者 rife-high

**Reply by zhongfly** [reply]
要紧跟更新，那不如直接fork仓库，然后合并你的修改，只不过得学着用git解决合并冲突就是了

**Reply by jiang-yuan** [reply]
mvtools-pot和rife-high 哪个比较好用啊
还有rife-2x就是rife-high 吗

**Reply by hooke007** [reply]
> 哪个比较好用啊

看你自己啊。。。喜欢哪个用哪个，跑得动哪个用哪个。以及我那篇rife帖子的几个图基本把两种补帧的最大特点写了

> rife-2x就是rife-high 吗

差不太多吧 https://github.com/hooke007/MPV_lazy/commit/7864a5f8f7115fd9b4954f7c3221a1da44e39edf#diff-e295864f83805611c7794449138e7d4b68e8a10cfa3b03a9ed69ddc24c9ed2c5

**Reply by jiang-yuan** [reply]
![Snipaste_2022-05-22_02-13-21](https://user-images.githubusercontent.com/36291537/169664232-1d1eb483-b6ed-4e94-b954-3ac08b62c155.png)
这滤镜有点狠,我笔记本1660ti都跑不动

**Reply by hooke007** [reply]
正常，对着说明调参吧

**Reply by jiang-yuan** [reply]
现在调的只有max_Leng 这个参数把?
就是说我如果是2160p的视频默认会给我缩到720p补帧了再放大?这样清晰度会下降把?

**Reply by hooke007** [reply]
不是还有新增加的 `pix_fmt2 = vs.YUV420P8` 吗

> 清晰度会下降

当然，不然我为什么唯独这个脚本里降分辨率，因为大多数人的显卡带不起

**Reply by jiang-yuan** [reply]
清晰度下降等于1080p以上的视频我不适合用它了,那就先不调了吧,我需要4k都能补帧的
之前用的是论坛找的这个配置,平时还够用,就是有时候人物周边一圈的背景会花
[svpflow-movie.txt](https://github.com/hooke007/MPV_lazy/files/8748143/svpflow-movie.txt)

**Reply by hooke007** [reply]
这不是我之前瞎写的吗。。。

**Reply by jiang-yuan** [reply]
哈哈要不要弄个改进版
目前默认配置真没几个能用真人电影的

**Reply by hooke007** [reply]
看了下是上古版本的产物
![Snipaste_2022-05-21_19-47-58](https://user-images.githubusercontent.com/41094733/169665360-d95a8ebf-d9dc-43f2-83e8-a1fc5a59004e.jpg)

> 真人电影

有点难搞，本身真人视频的运动风格就各有特色。偷懒的话就抄一抄manager自动生成的参数

pot那个应该是通用的，效果不行的话。。mvtools基本就只能那样了

**Reply by jiang-yuan** [reply]
manager自动生成的参数在哪
目前mvtools占cpu太多
svp两个配置都是2d的
rife又降分辨率
看个电影太难了

**Reply by hooke007** [reply]
> manager自动生成的参数在哪

![Snipaste_2022-05-21_19-56-10](https://user-images.githubusercontent.com/41094733/169665592-db1278e3-6e7d-4921-8c99-e490ca4485c4.jpg)

**Reply by hooke007** [reply]
看了下它自动生成的参数。。够简单的，动画和影视就靠algo（动画是2）区分
```
smoothfps_params = "{gpuid:11,gpu_qn:2,rate:{num:2,den:1},algo:13,mask:{area:100},scene:{}}"
```

**Reply by jiang-yuan** [reply]
明明是懒人包想看个电影结果还要去研究怎么写svp,头疼
如果直接安装svp它默认的配置对应mpvlazy配置的哪个啊

**Reply by hooke007** [reply]
svp用它自带的mpv，默认读的是 安装目录的 `mpv.conf` 和 `%APPDATA%/mpv/`
https://mpv.io/manual/master/#files-on-windows

**Reply by jiang-yuan** [reply]
我看恒星播放器的说明安装完svp就不用管了,播放器里有个svp的开关,但是没有配置的选择,挺好奇那些安装svp只用默认配置的效果怎么样

**Reply by hooke007** [reply]
一般人没这么多想法，，，能看就行

**Reply by letterk** [reply]
紧跟版本更新, 然后5-6个旧版本目录放在那里. 最初还要调一下input什么的, 现在都不动了

**Reply by ToumaLover** [reply]
可以麻烦在release里和文档里写一下正确的更新方式吗？为了找到你这句话开了十来个网页

> 正确的做法是，移除旧的，解压新的。

**Reply by hooke007** [reply]
release里那句话的意思是：不支持更新。

我们讨论的“更新“主要是指个人设置迁移。

对大多数便携软件来讲，移除旧的解压新的，就是干净且正确的更新方式。

**Reply by hooke007** [reply]
这个项目的早期阶段，脚本的变动十分频繁，有用户为了省删除这一步就直接解压覆盖旧文件。导致遗留的旧脚本或改名的脚本仍然在工作，自然产生了各种我无法复现的问题。

于是有了这个不要覆盖的提示。

对后期的版本来说，这个提示更加有存在的必要了，py的包和vs插件都容易存在依赖冲突。

当然措辞确实可以修改，防止被误以为支持自更新的功能。

**Comment by zhongfly** [comment]
也许可以用linux下diff然后patch的方法做？git bash里应该有这工具

windows下也有其他这类比较差异，合并差异的工具，可以试试

**Reply by jiang-yuan** [reply]
这个只适合增减行的情况吧
有些=后面的默认值新版也变了的话就会出两行赋值的语句了?
感觉文件会变得很乱

**Reply by zhongfly** [reply]
> 这个只适合增减行的情况吧 有些=后面的默认值新版也变了的话就会出两行赋值的语句了? 感觉文件会变得很乱

你了解下diff就知道了，diff得到的是差异变化，可不是只能什么增减行的，类似于git里的commit然后merge。你说的那种情况会直接patch失败，所以说建议你用文本对比工具

**Reply by jiang-yuan** [reply]
嗯,最后就打算用个对比工具一行行手动去调了,一点也不懒人

**Comment by jiang-yuan** [comment]
git解决冲突也是一条条人工选冲突选那个,其实也差不多




------------------&nbsp;Original&nbsp;------------------
From: ***@***.***&gt;; 
Date: 2022年4月27日(星期三) 凌晨0:20
To: ***@***.***&gt;; 
Cc: "Yuan ***@***.***&gt;; ***@***.***&gt;; 
Subject: Re: [hooke007/MPV_lazy] 有无比较方便的更新方案 (Discussion #117)





 
那不如直接fork仓库，然后合并你的修改，只不过得学着用git解决合并冲突就是了

**Reply by zhongfly** [reply]
如果只是mpv.conf这种设置文件的修改，我觉得随便写个脚本用正则之类的做就行…
搜索待修改的选项是否存在，存在则使用自己的设置覆盖该行，不存在则新建…可能powershell写一个就成
不过得考虑-append这类可重复的选项怎么处理…

**Comment by jiang-yuan** [comment]
没事已经用notepad++的对比插件对比完了
其实也就几个设置项,就是比较散自己记不住




------------------&nbsp;Original&nbsp;------------------
From: ***@***.***&gt;; 
Date: 2022年4月27日(星期三) 凌晨0:31
To: ***@***.***&gt;; 
Cc: "Yuan ***@***.***&gt;; ***@***.***&gt;; 
Subject: Re: [hooke007/MPV_lazy] 有无比较方便的更新方案 (Discussion #117)





 
如果只是mpv.conf这种设置文件的修改，我觉得随便写个脚本用正则之类的做就行…
 搜索待修改的选项是否存在，存在则使用自己的设置覆盖该行，不存在则新建…可能powershell写一个就成

**Reply by zhongfly** [reply]
你在邮件里直接回复的话，会新建一个answer，而不是像楼中楼那样引用回复

**Comment by jiang-yuan** [comment, marked as answer]
目前的解决方案是把自己的配置都写在最下面,这样能覆盖上面原本的设置而且原本的设置不用改动
下次更新直接从两个config文件里面复制最下面的设置到新版里面
